src/components/MessageInput/InputButtons.tsx

```diff
@@ -0,0 +1,215 @@
+  | 'text'
+  | 'uploadsEnabled'
+> & {
+  closeAttachmentPicker?: () => void;
+  openAttachmentPicker?: () => void;
+  openCommandsPicker?: () => void;
+  toggleAttachmentPicker?: () => void;
+};
+
+export const InputButtonsWithContext = <
+  At extends DefaultAttachmentType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+>(
+  props: InputButtonsWithContextProps<At, Ch, Co, Ev, Me, Re, Us>,
+) => {
+  const {
+    AttachButton,
+    CommandsButton,
+    giphyActive,
+    hasCommands,
+    hasFilePicker,
+    hasImagePicker,
+    MoreOptionsButton,
+    openCommandsPicker,
+    setShowMoreOptions,
+    showMoreOptions,
+    text,
+    toggleAttachmentPicker,
+    uploadsEnabled,
+  } = props;
+
+  const {
+    theme: {
+      messageInput: { attachButtonContainer, commandsButtonContainer },
+    },
+  } = useTheme();
+
+  if (giphyActive) {
+    return null;
+  }
+
+  return (
+    <>
+      {!showMoreOptions && (hasImagePicker || hasFilePicker) && hasCommands ? (
+        <MoreOptionsButton handleOnPress={() => setShowMoreOptions(true)} />
+      ) : (
+        <>
+          {(hasImagePicker || hasFilePicker) && uploadsEnabled !== false && (
+            <View
+              style={[
+                hasCommands ? styles.attachButtonContainer : undefined,
+                attachButtonContainer,
+              ]}
+            >
+              <AttachButton handleOnPress={toggleAttachmentPicker} />
+            </View>
+          )}
+          {hasCommands && !text && (
+            <View style={commandsButtonContainer}>
+              <CommandsButton handleOnPress={openCommandsPicker} />
+            </View>
+          )}
+        </>
+      )}
+    </>
+  );
+};
+const areEqual = <
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+>(
+  prevProps: InputButtonsWithContextProps<At, Ch, Co, Ev, Me, Re, Us>,
+  nextProps: InputButtonsWithContextProps<At, Ch, Co, Ev, Me, Re, Us>,
+) => {
+  if (prevProps.hasImagePicker !== nextProps.hasImagePicker) {
+    return false;
+  }
+
+  if (prevProps.hasFilePicker !== nextProps.hasFilePicker) {
+    return false;
+  }
+
+  if (prevProps.hasCommands !== nextProps.hasCommands) {
+    return false;
+  }
+
+  if (prevProps.uploadsEnabled !== nextProps.uploadsEnabled) {
+    return false;
+  }
+
+  if (prevProps.showMoreOptions !== nextProps.showMoreOptions) {
+    return false;
+  }
+
+  if (
+    (!prevProps.text && nextProps.text) ||
+    (prevProps.text && !nextProps.text)
```

Comment:
do nothing if the text changes? so for the case where a user already has text in the input and then clicks the button to expand the message input options then starts typing, nothing changes?

Reply:
`setHasMoreOptions` gets set from in `useMessageDetailsForState`, so thats a separate render cycle.